### PR TITLE
refactor:  surfgen template is updated

### PR DESCRIPTION
**Pull request by Evgenia-bit**
## Changes
- updated surfgen template files in tools/api_generator
- added surfgen script

## Checklist for self-check
- [x] Commits and PRs have been filed according to [the rules on the project](https://github.com/surfstudio/surf-flutter-app-template#workflow-in-a-repository).
- [x] The author is marked as an assigne and assigned mandatory reviewers.
- [x] Required labels marked
- [ ] Specified related tasks and/or related PRs.
- [x] Specified Changes.
- [ ] Attached videos/screenshots demonstrating the fix/feature.
- [x] All unspecified fields in the PR description deleted.
- [ ] New code covered by tests.

## Checklist for reviewers
- [ ] CI passed successfully _(with a green check mark)_.
- [ ] PR is atomic, by volume no more than 400 (+-) corrected lines (not including codogen).

Design:
- [ ] System design corresponds to the agreements on structure and architecture on the project.
- [ ] The code is decomposed into necessary and sufficient components.

Functionality:
- [ ] The code solves the problem.
- [ ] Any changes to the user interface are reasonable and look good.

Complexity:
- [ ] The code is clear, easy to read, functions are small, no more than 50 lines.
- [ ] The logic is not overcomplicated, there is no overengineering (no code sections that may be needed in the future, but no one knows about it).

Tests:
- [ ] Updated or added tests for mandatory components.
- [ ] The tests are correct, helpful, and well designed/developed.

Naming:
- [ ] The naming of variables, methods, classes and other components is understandable.

Comments:
- [ ] The comments are understandable and helpful.

Documentation:
- [ ] All labels are correct
- [ ] Technical documentation updated (after approval, updates last reviewer).

### CODE REVIEW

File: scripts/surfgen.sh [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/env bash
+
+if [ "$(basename "$PWD")" = 'script' ]; then cd ..; fi
```
Comment (dzmitry-struk-surf) [inline scripts/surfgen.sh:3]:
```suggestion
if [ "$(basename "$PWD")" = 'scripts' ]; then cd ..; fi
```

File: tools/api_generator/dto_codegen.sh [hunk 2]
```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env bash
+
+data_output_path=../../lib/api/data
+umbrella_file_name=umbrella.dart
+
+### Generate swagger configuration
+for i in $(find ../../../flutter-template-swagger/**/api.yaml); do
```
Comment (dzmitry-struk-surf) [inline tools/api_generator/dto_codegen.sh:7]:
```suggestion
### Generate swagger configuration
### TODO(init-project): Replace "flutter-template-swagger" with your project name
for i in $(find ../../../flutter-template-swagger/**/api.yaml); do
```

File: tools/api_generator/dto_codegen.sh [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+### Create new umbrella file
+for value in "${fileNames[@]}"
+do
+  name=${value%.*}
+  echo "export '$name.dart';" >> "$umbrella_file_name"
+done
+
+### Format all generated files
+fvm dart format -l 100 lib packages test
```
Comment (dzmitry-struk-surf) [inline tools/api_generator/dto_codegen.sh:40]:
```suggestion
### TODO(init-project): change to your project line length and desired folders for formatting
fvm dart format -l 120 lib packages test
```

File: tools/api_generator/dto_config.yaml [hunk 4]
```diff
@@ -0,0 +1,15 @@
+useNewNullableDeterminationStrategy: true
+
+templates:
+  - type: model
+    fileExtension: dart
+    nameSuffix: Data
```
Comment (dzmitry-struk-surf) [inline tools/api_generator/dto_config.yaml:6]:
```suggestion
    nameSuffix: Dto
```

File: tools/api_generator/dto_config.yaml [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+    fileNameCase: snakeCase
+    templatePath: templates/retrofit_model.stencil
+    destinationPath: ../../lib/api/data
+  - type: enum
+    fileExtension: dart
+    nameSuffix: Data
```
Comment (dzmitry-struk-surf) [inline tools/api_generator/dto_config.yaml:12]:
```suggestion
    nameSuffix: Dto
```

File: tools/api_generator/request_codegen.sh [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+done
+
+### Move to project folder
+cd ../..
+
+### Generate all *.g.dart files
+fvm flutter clean && fvm flutter pub get && fvm flutter pub run build_runner build --delete-conflicting-outputs
+
+### Format all generated files
+fvm dart format -l 100 lib packages test
```
Comment (dzmitry-struk-surf) [inline tools/api_generator/request_codegen.sh:15]:
```suggestion
### Format all generated files
### TODO(init-project): change to your project line length and desired folders for formatting
fvm dart format -l 120 lib packages test
```

File: tools/api_generator/request_codegen.sh [hunk 7]
```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env bash
+
+### Generate swagger configuration
+for i in $(find ../../../flutter-template-swagger/**/api.yaml); do
```
Comment (dzmitry-struk-surf) [inline tools/api_generator/request_codegen.sh:4]:
```suggestion
### Generate swagger configuration
### TODO(init-project): Replace "flutter-template-swagger" with your project name
for i in $(find ../../../flutter-template-swagger/**/api.yaml); do
```

File: tools/api_generator/templates/model.stencil
Comment (dzmitry-struk-surf) [inline tools/api_generator/templates/model.stencil:1]: This file is not used, it can be removed.

File: tools/api_generator/dto_config.yaml [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+    fileNameCase: snakeCase
+    templatePath: templates/enum_retrofit.stencil
```
Comment (dzmitry-struk-surf) [inline tools/api_generator/dto_config.yaml:14]: This template is missing in `templates` folder. It have to be added.

**Review by dzmitry-struk-surf — CHANGES_REQUESTED** [review]

File: tools/api_generator/dto_codegen.sh (hunk 3 above)
Comment (Evgenia-bit) [reply tools/api_generator/dto_codegen.sh:40]: Right now, in our project, the line length is 100 everywhere, can we leave 100 here?

File: tools/api_generator/dto_config.yaml (hunk 8 above)
Comment (Evgenia-bit) [reply tools/api_generator/dto_config.yaml:14]: He is there:
Screenshot 2024-02-29 at 14 42 36

Comment (Evgenia-bit) [reply tools/api_generator/dto_config.yaml:14]: Maybe it 's better to name it `retrofit_enum.stencil`?

File: tools/api_generator/dto_codegen.sh (hunk 2 above)
Comment (dzmitry-struk-surf) [reply tools/api_generator/dto_codegen.sh:7]: Yep, agree.

File: tools/api_generator/dto_codegen.sh (hunk 3 above)
Comment (dzmitry-struk-surf) [reply tools/api_generator/dto_codegen.sh:40]: We will change to 120 later. I will write it do to not to forget. Yep, let's leave it 100 for now.

File: tools/api_generator/dto_config.yaml (hunk 8 above)
Comment (dzmitry-struk-surf) [reply tools/api_generator/dto_config.yaml:14]: Yep, agree. Let's rename it.

**Review by dzmitry-struk-surf — APPROVED** [review]

**Pull request merged**